client/login.go

```diff
@@ -219,3 +220,109 @@ func RefreshToken(data RefreshTokenData) {
 	}
 	TasksMux.Unlock()
 }
+
+// OAuth2LogoutHandler handles the request signaling that the user has logged out.
+// It devalidates the
+func OAuth2LogoutHandler(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Error reading request body: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer r.Body.Close()
+
+	var data RefreshTokenData
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		http.Error(w, "Error parsing JSON: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	go OAuth2Logout(data)
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// OAuth2Logout sends revocation request to the server to revoke the tokens.
+// It logs out the user from all add-ons.
+func OAuth2Logout(data RefreshTokenData) {
+	var wg sync.WaitGroup
+	var ch = make(chan error, 2)
+	wg.Add(2)
+	go RevokeOAuth2Token(data, "api_key", ch, &wg)
+	go RevokeOAuth2Token(data, "refresh_token", ch, &wg)
+	wg.Wait()
+	close(ch)
+	var errors []error
+	for err := range ch {
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	var message, status string
+	if len(errors) == 0 {
+		message = "Logout OK, tokens successfully revoked on the server"
+		status = "finished"
+	} else if len(errors) == 1 {
+		message = fmt.Sprintf("Logout partially OK, failed to revoke token: %v", errors[0])
+		status = "error"
+	} else {
+		message = fmt.Sprintf("Logout partially OK, failed to revoke tokens: %v", errors)
+		status = "error"
+	}
+
+	for appID := range Tasks {
+		task := NewTask(data, appID, uuid.New().String(), "oauth2/logout")
+		task.Message = message
+		task.Status = status
+		task.Error = fmt.Errorf(message) // just to print it into console
+		AddTaskCh <- task
+	}
+}
+
+// RevokeOAuth2Token revokes api_key or refresh_token according to RFC 7009: https://www.rfc-editor.org/rfc/rfc7009.html#section-2.1.
+// Token type is either "api_key" or "refresh_token".
+func RevokeOAuth2Token(data RefreshTokenData, tokenType string, ch chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	rData := url.Values{}
+	rData.Set("client_id", OAUTH_CLIENT_ID)
+	if tokenType == "api_key" {
+		rData.Set("token", data.APIKey)
+		rData.Set("token_type_hint", "refresh_token")
```

Comment:
This should work but this:
```suggestion
```
or this:
```suggestion
		rData.Set("token_type_hint", "access_token")
```
is safer.

Reply:
@radekholy24 Ha, jsem to blbe pochopil s tim bugem ohledne type hint. Tedy jsem to ted odstranil a bezi to dobre, diky za vsimnuti!